From 68c81bbea57e637f0382a94b01160431697fa1b2 Mon Sep 17 00:00:00 2001
From: Felipe Balbi <balbi@ti.com>
Date: Thu, 21 Nov 2013 12:30:31 -0600
Subject: [PATCH 133/285] arm: dts: am43xx: add ADC for am437x-gp and
 am43x-epos

Add ADC support for both GP EVM and EPOS EVM.

Signed-off-by: Felipe Balbi <balbi@ti.com>
[Kevin: The original patch taken from TI
ti-sdk-am437x-evm-07.01.00.00-Linux-x86-Install.bin]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm/boot/dts/am4372.dtsi       |    1 +
 arch/arm/boot/dts/am437x-gp-evm.dts |    8 ++++++++
 2 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/am4372.dtsi b/arch/arm/boot/dts/am4372.dtsi
index dfcf10a..68f749a 100644
--- a/arch/arm/boot/dts/am4372.dtsi
+++ b/arch/arm/boot/dts/am4372.dtsi
@@ -786,6 +786,7 @@
 			reg = <0x44e0d000 0x1000>;
 			ti,hwmods = "adc_tsc";
 			clocks = <&adc_tsc_fck>;
+			interrupts = <GIC_SPI 16 IRQ_TYPE_LEVEL_HIGH>;
 			clock-names = "fck";
 			status = "disabled";
 
diff --git a/arch/arm/boot/dts/am437x-gp-evm.dts b/arch/arm/boot/dts/am437x-gp-evm.dts
index 99343c7..10daba4 100644
--- a/arch/arm/boot/dts/am437x-gp-evm.dts
+++ b/arch/arm/boot/dts/am437x-gp-evm.dts
@@ -469,6 +469,14 @@
 	};
 };
 
+&tscadc {
+	status = "okay";
+
+	adc {
+		ti,adc-channels = <0 1 2 3 4 5 6 7>;
+	};
+};
+
 &ecap0 {
 	status = "okay";
 	pinctrl-names = "default";
-- 
1.7.5.4

